signing_service/src/tests/tests.py

```diff
@@ -0,0 +1,160 @@
+from unittest import TestCase
+import socket
+import sys
+
+import mock
+
+from ..constants import SIGNING_SERVICE_DEFAULT_PORT
+from ..constants import SIGNING_SERVICE_RECOVERABLE_ERRORS
+from ..constants import SIGNING_SERVICE_MAXIMUM_RECONNECT_TIME
+from ..signing_service import _parse_arguments
+from ..signing_service import SigningService
+
+
+class SigningServiceMainTestCase(TestCase):
+
+    def setUp(self):
+        self.host = '127.0.0.1'
+        self.port = 8000
+        self.initial_reconnect_delay = 2
+
+    def test_that_signing_service_should_be_instantiated_correctly_with_all_parameters(self):
+        signing_service = SigningService(self.host, self.port, self.initial_reconnect_delay)
+
+        self.assertIsInstance(signing_service, SigningService)
+        self.assertEqual(signing_service.host, self.host)
+        self.assertEqual(signing_service.port, self.port)
+        self.assertEqual(signing_service.initial_reconnect_delay, self.initial_reconnect_delay)
+
+    def test_that_signing_service_should_run_full_loop_when_be_instantiated_with_all_parameters(self):
+        with mock.patch('socket.socket.connect') as mock_socket_connect:
+            with mock.patch('src.signing_service.SigningService._handle_connection') as mock__handle_connection:
```

Comment:
double underscore in: `mock__handle_connection`

Reply:
That is the name of the method :)